drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -507,8 +507,10 @@ local function device_added(driver, device)
   device:emit_event(capabilities.tamperAlert.tamper.clear())
   local eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.DoorLockFeature.PIN_CREDENTIALS})
   if #eps == 0 then
-    device.log.debug("Device does not support lockCodes")
-    device:try_update_metadata({profile = "lock-without-codes"})
+    if device:supports_capability_by_id(capabilities.tamperAlert.ID) then
+      device.log.debug("Device does not support lockCodes")
+      device:try_update_metadata({profile = "lock-without-codes"})
+    end
```

Comment:
What do you think about adding an else with a warning that we don't have a profile for a device that doesn't support codes or tamper?

Reply:
I think this is only temporary until the incoming profile gets merged. Then we can actually do switching in that `else`

Reply:
updated anyway, though